src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -62,4 +63,8 @@ def projects_for_filter
            .order('name desc').group(:name).count # this query returns a sorted-by-name hash like { "home:b" => 1, "home:a" => 3  }
            .sort_by(&:last).reverse.to_h # this sorts the hash by amount: { "home:a" => 3, "home:b" => 1 }
   end
+
+  def notifications_count
+    Notification.where(subscriber: User.session, web: true, delivered: false).group(:notifiable_type).count
```

Comment:
I suggest using [NotificationsFinder](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/queries/notifications_finder.rb) to re-use what we already have.

This would then be:

`NotificationsFinder.new(User.session.notifications.for_web).unread.group(:notifiable_type).count`

Reply:
@dmarcoux I addressed it, please check again :) also @DavidKang as you suggested, I'm adding the amount of unread notifications to the hash now, instead of doing a separate call in the view